test/IonTimestamp.ts

```diff
@@ -242,4 +242,11 @@ describe("Timestamp", () => {
         assert.deepEqual(ts.getSecondsDecimal(), ion.Decimal.parse('45'));
         assert.deepEqual(ts._getFractionalSeconds(), ion.Decimal.ZERO);
     });
+    it('constructor with Date', () => {
+        const date = new Date(2000, 0, 1, 12, 30, 45);
+        let timestamp1 = new ion.Timestamp(-7 * 60, 2000, 1, 1, 12, 30, 45);
+        let timestamp2 = ion.Timestamp.fromDate(date);
```

Comment:
The [Ion spec specifies](https://amzn.github.io/ion-docs/docs/spec.html#timestamp) that two timestamps are only equal if (among other things) their offsets are the same:

> Two timestamps are only equivalent if they represent the same instant with **the same offset** and precision.

(Emphasis mine.)

`timestamp1` specifies an offset of UTC-7 while `timestamp2` is built from a source `Date` that doesn't specify an offset. The `Date`'s offset will be that of the host system, which will vary between developer machines and build servers. This means that the two timestamps will only be considered equal when the test is run on a machine with an offset of UTC-7.

`Date`'s constructor doesn't allow you to specify a timezone offset directly (weird!), and one might reasonably be tempted to provide one by passing an ISO8601 timestamp instead of individual parameters. Unfortunately, JavaScript will only adjust the resulting `Date` back to the host system's local time:

![image](https://user-images.githubusercontent.com/611616/106768011-7fcfdd80-6609-11eb-8c01-8a020bbb7838.png)

This means that there's not going to be a reliable way to construct a `Date` with the desired offset and configure `timestamp2` with the same offset independently. I think the best we can do is something like this:

```suggestion
        const date = new Date(2000, 0, 1, 12, 30, 45);
        let timestamp1 = new ion.Timestamp(date.getTimezoneOffset() * -1, 2000, 1, 1, 12, 30, 45);
        let timestamp2 = ion.Timestamp.fromDate(date);
```

I was pleased to learn this morning that TC39, the standards body that governs ECMAScript/JavaScript, [is working on a new datetime standard](https://tc39.es/proposal-temporal/docs/) that looks to be much more sane.  🙂